src/Helper.php

```diff
@@ -37,23 +56,25 @@ protected static function output(File $model, array $headers, App $app = null):
     {
         // TODO move to App and add support for streams
 
-        $headers = self::normalizeHeaders($headers);
-
         $location = $model->get('location');
 
         if ($app !== null) {
-            $app->terminate($model->flysystem->read($location), $headers);
+            foreach ($headers as $name => $value) {
+                $app->setResponseHeader($name, $value);
+            }
+            $app->terminate($model->flysystem->read($location));
         }
 
         $isCli = \PHP_SAPI === 'cli'; // for phpunit
 
-        foreach ($headers as $k => $v) {
+        $headers = self::normalizeHeaders($headers);
```

Comment:
this is needs to be reworked to utilize https://github.com/atk4/ui/pull/1706

I have some changes on my work computer, but I am very busy these days...

Reply:
Ok. I added few other fixes here and also included your https://github.com/atk4/filestore/pull/32
Really would love to merge this in so we have working filestore again and you can add streaming support later on when you have more time in separate PR. Is that OK?

Reply:
The streaming support should be easy to add, so I prefer #32 to be finished to really "Adjust to atk4/ui PSR-7 response" implement as the PR title claims. Feel free to send a PR againts my PR branch.

Reply:
It already uses `$app->setResponseHeader` and `$app->terminate` without additional header array.
I don't get what else do you want here?